lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeMultiSegmentTest.kt

```diff
@@ -0,0 +1,544 @@
+            ),
+            NucleotideSymbolEquals("seg1", 300, "G"),
+        )
+
+        assertThat(result, equalTo(expectedResult))
+    }
+
+    @Test
+    fun `given a complex advanced query with MAYBE then map should return the corresponding SiloQuery`() {
+        val advancedQuery =
+            "MAYBE((seg1:700B | seg1:800-) & !seg2:600 & [3-of: seg1:123A, seg2:234T, seg1:345G]) & " +
+                "pangoLineage=jn.1* & some_metadata.regex='^Democratic.*'"
+
+        val result = underTest.map(advancedQuery)
+
+        val expectedResult = And(
+            StringSearch("some_metadata", "^Democratic.*"),
+            LineageEquals(PANGO_LINEAGE_COLUMN, "jn.1", true),
+            Maybe(
+                And(
+                    NOf(
+                        3,
+                        matchExactly = false,
+                        listOf(
+                            NucleotideSymbolEquals("seg1", 123, "A"),
+                            NucleotideSymbolEquals("seg2", 234, "T"),
+                            NucleotideSymbolEquals("seg1", 345, "G"),
+                        ),
+                    ),
+                    Not(HasNucleotideMutation("seg2", 600)),
+                    Or(
+                        NucleotideSymbolEquals("seg1", 800, "-"),
+                        NucleotideSymbolEquals("seg1", 700, "B"),
+                    ),
+                ),
+            ),
+        )
+
+        assertThat(result, equalTo(expectedResult))
+    }
+
+    @Test
+    fun `given a advancedQuery with an invalid nucleotide mutation then throw error`() {
+        val aaMutation = "seg1:300P"
+
+        val exceptionAa = assertThrows<BadRequestException> { underTest.map(aaMutation) }
+        assertThat(exceptionAa.message, `is`("Invalid nucleotide symbol: P"))
+
+        val fromAaMutation = "seg1:R300G"
+        val exception = assertThrows<BadRequestException> { underTest.map(fromAaMutation) }
+        assertThat(exception.message, `is`("Invalid nucleotide symbol: R"))
+    }
+
+    @Test
+    fun `given a advancedQuery with an invalid amino acid mutation then throw error`() {
+        val nucMutation = "S:300B"
+
+        val exceptionNuc = assertThrows<BadRequestException> { underTest.map(nucMutation) }
+        assertThat(exceptionNuc.message, `is`("Invalid amino acid symbol: B"))
+    }
+
+    @Test
+    fun `given a advancedQuery with an invalid nucleotide insertion then throw error`() {
+        val aaMutation = "ins_seg1:300:GP"
+
+        val exceptionAa = assertThrows<BadRequestException> { underTest.map(aaMutation) }
+        assertThat(exceptionAa.message, `is`("Invalid nucleotide symbol: P"))
+    }
+
+    @Test
+    fun `given a advancedQuery with an invalid amino acid insertion then throw error`() {
+        val nucMutation = "ins_S:300:GB"
+
+        val exceptionNuc = assertThrows<BadRequestException> { underTest.map(nucMutation) }
+        assertThat(exceptionNuc.message, `is`("Invalid amino acid symbol: B"))
+    }
```

Comment:
This would also profit quite a bit from using parametrized tests (also the valid cases).

Reply:
I've spent a lot of time changing the other tests to be parametrized and I don't think it is worth the time.